onnxruntime/core/framework/fallback_cpu_capability.cc

```diff
@@ -90,8 +93,11 @@ std::unordered_set<NodeIndex> GetCpuPreferredNodes(const onnxruntime::GraphViewe
         }));
   }
 
-  const std::vector<const NodeArg*>& graph_inputs = graph.GetInputs();
-  std::unordered_set<NodeIndex> cpu_nodes;
+  gsl::span<const NodeArg* const> graph_inputs = graph.GetInputs();
+  InlinedHashSet<NodeIndex> visited;
+  visited.reserve(candidates.size());
```

Comment:
>candidates.size() [](http://example.com/codeflow?start=18&length=17)

I think this size may not be big enough - candidates gets output node indexes added to it within the loop